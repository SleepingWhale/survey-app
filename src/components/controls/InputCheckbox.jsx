import React, { Component } from 'react';
import PropTypes from 'prop-types';

export class InputCheckbox extends Component {
  static propTypes = {
    id: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
    value: PropTypes.bool,
    question: PropTypes.string
  };

  static defaultProps = {
    value: false,
    question: ''
  };

  onChange = e => {
    const { onChange } = this.props;
    const { checked } = e.target;
    onChange(checked);
  };

  render() {
    const { value, question, id } = this.props;

    return (
      <div className="form-group">
        <div className="form-check">
          <input
            type="checkbox"
            className="form-check-input"
            id={id}
            checked={value}
            onChange={this.onChange}
          />
          <label className="form-check-label" htmlFor={id}>
            {question}
          </label>
        </div>
      </div>
    );
  }
}
