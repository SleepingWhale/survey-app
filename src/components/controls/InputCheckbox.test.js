import React from 'react';
import { mount } from 'enzyme';
import { InputCheckbox } from './InputCheckbox';

describe('InputCheckbox', () => {
  it('should render correctly', () => {
    const onChangeHandler = jest.fn();
    const component = mount(
      <InputCheckbox
        onChange={onChangeHandler}
        value={false}
        id="id"
        question="question"
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should call handler', () => {
    const onChangeHandler = jest.fn();
    const component = mount(
      <InputCheckbox
        onChange={onChangeHandler}
        value={false}
        id="id"
        question="question"
      />
    );
    const input = component.find('input').at(0);

    input.simulate('change', { target: { checked: true } });

    expect(onChangeHandler).toHaveBeenNthCalledWith(1, true);
  });
});
