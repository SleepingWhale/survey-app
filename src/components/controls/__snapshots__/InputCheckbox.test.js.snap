// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputCheckbox should render correctly 1`] = `
<InputCheckbox
  id="id"
  onChange={[MockFunction]}
  question="question"
  value={false}
>
  <div
    className="form-group"
  >
    <div
      className="form-check"
    >
      <input
        checked={false}
        className="form-check-input"
        id="id"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="form-check-label"
        htmlFor="id"
      >
        question
      </label>
    </div>
  </div>
</InputCheckbox>
`;
